### ISSUE TITLE
`ccache` from `pkgs.pkgsCross.gnu32` fails tests on a x86-64 host

### ISSUE THREAD

**Issue by peperunas** [NixOS/nixpkgs#149880]
### Describe the bug
`ccache` fails to build while cross-compiling on a `x86-64` host from `pkgsCross.gnu32`.

### Steps To Reproduce
1. Use `ccache` from `pkgs.pkgsCross.gnu32` on a `x86-64` host

### Additional context
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/irw6ab6q2aqf631b5rw0s2z02abnx6si-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/71lpqnlrp2gwy7nnckknq5iidq677sgk-force-objdump-on-darwin.patch
patching file test/run
Hunk #1 succeeded at 148 (offset 22 lines).
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_LOCALEDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/share/doc/ccache -DCMAKE_INSTALL_INFODIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/qjcvlj15x3zz8jngwn777acrsah8zkvh-ccache-4.5.1-man/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/js66s0xwjnzg0ggi2lq9bcvlk6x2za13-binutils-2.35.2/bin/ranlib -DCMAKE_AR=/nix/store/js66s0xwjnzg0ggi2lq9bcvlk6x2za13-binutils-2.35.2/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/5xrvbk440ln0nni4rq81jz4g736fmlkg-ccache-4.5.1 -DREDIS_STORAGE_BACKEND=OFF 
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/gcc
-- Ccache version: 4.5.1
-- Ccache dev mode: OFF
-- Performing Test std_atomic_without_libatomic
-- Performing Test std_atomic_without_libatomic - Success
-- Looking for linux/fs.h
-- Looking for linux/fs.h - found
-- Looking for pwd.h
-- Looking for pwd.h - found
-- Looking for sys/clonefile.h
-- Looking for sys/clonefile.h - not found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - found
-- Looking for sys/file.h
-- Looking for sys/file.h - found
-- Looking for syslog.h
-- Looking for syslog.h - found
-- Looking for termios.h
-- Looking for termios.h - found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for utime.h
-- Looking for utime.h - found
-- Looking for sys/utime.h
-- Looking for sys/utime.h - not found
-- Looking for varargs.h
-- Looking for varargs.h - not found
-- Looking for asctime_r
-- Looking for asctime_r - found
-- Looking for geteuid
-- Looking for geteuid - found
-- Looking for getopt_long
-- Looking for getopt_long - found
-- Looking for getpwuid
-- Looking for getpwuid - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - found
-- Looking for realpath
-- Looking for realpath - found
-- Looking for setenv
-- Looking for setenv - found
-- Looking for strndup
-- Looking for strndup - found
-- Looking for syslog
-- Looking for syslog - found
-- Looking for unsetenv
-- Looking for unsetenv - found
-- Looking for utimes
-- Looking for utimes - found
-- Performing Test HAVE_PTHREAD_MUTEX_ROBUST
-- Performing Test HAVE_PTHREAD_MUTEX_ROBUST - Success
-- Looking for pthread_mutexattr_setpshared
-- Looking for pthread_mutexattr_setpshared - found
-- Performing Test HAVE_STRUCT_STAT_ST_ATIM
-- Performing Test HAVE_STRUCT_STAT_ST_ATIM - Success
-- Performing Test HAVE_STRUCT_STAT_ST_CTIM
-- Performing Test HAVE_STRUCT_STAT_ST_CTIM - Success
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM - Success
-- Performing Test HAVE_STRUCT_STATFS_F_FSTYPENAME
-- Performing Test HAVE_STRUCT_STATFS_F_FSTYPENAME - Failed
-- Performing Test HAVE_AVX2
-- Performing Test HAVE_AVX2 - Success
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long long
-- Check size of long long - done
-- Found zstd: /nix/store/nbj4yv05zp8h65lqjn4w5vylpydynm13-zstd-1.5.0-dev/include (Required is at least version "1.1.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test HAVE_ASM_SSE2
-- Performing Test HAVE_ASM_SSE2 - Success
-- Performing Test HAVE_ASM_SSE41
-- Performing Test HAVE_ASM_SSE41 - Success
-- Performing Test HAVE_ASM_AVX2
-- Performing Test HAVE_ASM_AVX2 - Success
-- Performing Test HAVE_ASM_AVX512
-- Performing Test HAVE_ASM_AVX512 - Success
-- Performing Test HAVE_NEON
-- Performing Test HAVE_NEON - Failed
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j24 -l24 SHELL=/nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-bash-5.1-p8/bin/bash
Scanning dependencies of target blake3
[  1%] Generating ccache.1.tmp
[  2%] Building C object src/third_party/CMakeFiles/third_party.dir/base32hex.c.o
[  3%] Building CXX object src/third_party/CMakeFiles/third_party.dir/httplib.cpp.o
[  3%] Building CXX object src/third_party/CMakeFiles/third_party.dir/format.cpp.o
[  4%] Building CXX object src/third_party/CMakeFiles/third_party.dir/url.cpp.o
[  5%] Building C object src/third_party/CMakeFiles/third_party.dir/xxhash.c.o
[  6%] Building C object src/third_party/CMakeFiles/third_party.dir/getopt_long.c.o
[  7%] Building C object src/third_party/CMakeFiles/third_party.dir/xxh_x86dispatch.c.o
[  8%] Generating AUTHORS.html
[  9%] Building C object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_portable.c.o
[ 10%] Generating LICENSE.html
[ 11%] Building C object src/third_party/blake3/CMakeFiles/blake3.dir/blake3.c.o
[ 11%] Generating MANUAL.html
[ 12%] Generating NEWS.html
[ 13%] Building C object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_dispatch_ccache.c.o
[ 13%] Building ASM object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_avx2_x86-64_unix.S.o
[ 14%] Building ASM object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_avx512_x86-64_unix.S.o
[ 14%] Building ASM object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_sse41_x86-64_unix.S.o
[ 15%] Building ASM object src/third_party/blake3/CMakeFiles/blake3.dir/blake3_sse2_x86-64_unix.S.o
[ 16%] Generating ccache.1
[ 16%] Built target doc-man-page
[ 16%] Built target doc-html
[ 16%] Built target doc
[ 17%] Linking C static library libblake3.a
[ 17%] Built target blake3
[ 18%] Linking CXX static library libthird_party.a
[ 18%] Built target third_party
[ 19%] Building CXX object src/CMakeFiles/ccache_framework.dir/Args.cpp.o
[ 19%] Building CXX object src/CMakeFiles/ccache_framework.dir/Config.cpp.o
[ 20%] Building CXX object src/CMakeFiles/ccache_framework.dir/AtomicFile.cpp.o
[ 21%] Building CXX object src/CMakeFiles/ccache_framework.dir/Depfile.cpp.o
[ 22%] Building CXX object src/CMakeFiles/ccache_framework.dir/Context.cpp.o
[ 23%] Building CXX object src/CMakeFiles/ccache_framework.dir/Fd.cpp.o
[ 24%] Building CXX object src/CMakeFiles/ccache_framework.dir/Lockfile.cpp.o
[ 25%] Building CXX object src/CMakeFiles/ccache_framework.dir/Hash.cpp.o
[ 25%] Building CXX object src/CMakeFiles/ccache_framework.dir/Logging.cpp.o
[ 26%] Building CXX object src/CMakeFiles/ccache_framework.dir/Manifest.cpp.o
[ 27%] Building CXX object src/CMakeFiles/ccache_framework.dir/ProgressBar.cpp.o
[ 28%] Building CXX object src/CMakeFiles/ccache_framework.dir/Result.cpp.o
[ 30%] Building CXX object src/CMakeFiles/ccache_framework.dir/ResultRetriever.cpp.o
[ 30%] Building CXX object src/CMakeFiles/ccache_framework.dir/ResultDumper.cpp.o
[ 31%] Building CXX object src/CMakeFiles/ccache_framework.dir/SignalHandler.cpp.o
[ 31%] Building CXX object src/CMakeFiles/ccache_framework.dir/ResultExtractor.cpp.o
[ 32%] Building CXX object src/CMakeFiles/ccache_framework.dir/Stat.cpp.o
[ 33%] Building CXX object src/CMakeFiles/ccache_framework.dir/TemporaryFile.cpp.o
[ 34%] Building CXX object src/CMakeFiles/ccache_framework.dir/ThreadPool.cpp.o
[ 35%] Building CXX object src/CMakeFiles/ccache_framework.dir/Util.cpp.o
[ 36%] Building CXX object src/CMakeFiles/ccache_framework.dir/argprocessing.cpp.o
[ 36%] Building CXX object src/CMakeFiles/ccache_framework.dir/assertions.cpp.o
[ 37%] Building CXX object src/CMakeFiles/ccache_framework.dir/compopt.cpp.o
[ 38%] Building CXX object src/CMakeFiles/ccache_framework.dir/ccache.cpp.o
[ 39%] Building CXX object src/CMakeFiles/ccache_framework.dir/execute.cpp.o
[ 40%] Building CXX object src/CMakeFiles/ccache_framework.dir/hashutil.cpp.o
[ 41%] Building CXX object src/CMakeFiles/ccache_framework.dir/language.cpp.o
[ 41%] Building CXX object src/CMakeFiles/ccache_framework.dir/version.cpp.o
[ 42%] Building CXX object src/CMakeFiles/ccache_framework.dir/InodeCache.cpp.o
[ 43%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/Compressor.cpp.o
[ 44%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/Decompressor.cpp.o
[ 45%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/NullCompressor.cpp.o
[ 46%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/NullDecompressor.cpp.o
[ 46%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/ZstdCompressor.cpp.o
[ 47%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/ZstdDecompressor.cpp.o
[ 48%] Building CXX object src/CMakeFiles/ccache_framework.dir/compression/types.cpp.o
[ 50%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/CacheEntryHeader.cpp.o
[ 51%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/CacheEntryReader.cpp.o
[ 52%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/CacheEntryWriter.cpp.o
[ 53%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/Statistics.cpp.o
[ 53%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/StatisticsCounters.cpp.o
[ 54%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/StatsLog.cpp.o
[ 55%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/mainoptions.cpp.o
[ 56%] Building CXX object src/CMakeFiles/ccache_framework.dir/core/types.cpp.o
[ 57%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/CacheFile.cpp.o
[ 58%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/PrimaryStorage.cpp.o
[ 58%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/PrimaryStorage_cleanup.cpp.o
[ 59%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/PrimaryStorage_compress.cpp.o
[ 60%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/PrimaryStorage_statistics.cpp.o
[ 61%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/StatsFile.cpp.o
[ 62%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/primary/util.cpp.o
[ 63%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/secondary/FileStorage.cpp.o
[ 63%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/secondary/HttpStorage.cpp.o
[ 64%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/secondary/SecondaryStorage.cpp.o
[ 65%] Building CXX object src/CMakeFiles/ccache_framework.dir/storage/Storage.cpp.o
[ 66%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/TextTable.cpp.o
[ 67%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/Tokenizer.cpp.o
[ 68%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/file.cpp.o
[ 68%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/path.cpp.o
[ 69%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/string.cpp.o
[ 70%] Linking CXX static library libccache_framework.a
[ 70%] Built target ccache_framework
[ 71%] Building CXX object CMakeFiles/ccache.dir/src/main.cpp.o
[ 71%] Building CXX object unittest/CMakeFiles/unittest.dir/TestUtil.cpp.o
[ 73%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Args.cpp.o
[ 73%] Building CXX object unittest/CMakeFiles/unittest.dir/main.cpp.o
[ 74%] Building CXX object unittest/CMakeFiles/unittest.dir/test_AtomicFile.cpp.o
[ 74%] Building CXX object unittest/CMakeFiles/unittest.dir/test_FormatNonstdStringView.cpp.o
[ 75%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Config.cpp.o
[ 76%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Depfile.cpp.o
[ 77%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Lockfile.cpp.o
[ 78%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Hash.cpp.o
[ 79%] Building CXX object unittest/CMakeFiles/unittest.dir/test_NullCompression.cpp.o
[ 80%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Stat.cpp.o
[ 81%] Building CXX object unittest/CMakeFiles/unittest.dir/test_Util.cpp.o
[ 81%] Building CXX object unittest/CMakeFiles/unittest.dir/test_argprocessing.cpp.o
[ 82%] Building CXX object unittest/CMakeFiles/unittest.dir/test_ZstdCompression.cpp.o
[ 85%] Building CXX object unittest/CMakeFiles/unittest.dir/test_compression_types.cpp.o
[ 85%] Building CXX object unittest/CMakeFiles/unittest.dir/test_ccache.cpp.o
[ 85%] Building CXX object unittest/CMakeFiles/unittest.dir/test_compopt.cpp.o
[ 87%] Building CXX object unittest/CMakeFiles/unittest.dir/test_core_StatisticsCounters.cpp.o
[ 87%] Building CXX object unittest/CMakeFiles/unittest.dir/test_core_Statistics.cpp.o
[ 87%] Building CXX object unittest/CMakeFiles/unittest.dir/test_core_StatsLog.cpp.o
[ 90%] Building CXX object unittest/CMakeFiles/unittest.dir/test_storage_primary_util.cpp.o
[ 90%] Building CXX object unittest/CMakeFiles/unittest.dir/test_storage_primary_StatsFile.cpp.o
[ 90%] Building CXX object unittest/CMakeFiles/unittest.dir/test_hashutil.cpp.o
[ 91%] Linking CXX executable ccache
[ 91%] Built target ccache
[ 92%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_TextTable.cpp.o
[ 93%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_Tokenizer.cpp.o
[ 93%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_XXH3_128.cpp.o
[ 94%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_XXH3_64.cpp.o
[ 95%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_expected.cpp.o
[ 96%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_path.cpp.o
[ 97%] Building CXX object unittest/CMakeFiles/unittest.dir/test_util_string.cpp.o
[ 98%] Building CXX object unittest/CMakeFiles/unittest.dir/test_InodeCache.cpp.o
[100%] Linking CXX executable unittest
[100%] Built target unittest
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
Test project /build/source/build
      Start  1: unittest
      Start  2: test.base
      Start  3: test.basedir
      Start  4: test.cache_levels
      Start  5: test.cleanup
      Start  6: test.color_diagnostics
      Start  7: test.config
      Start  8: test.cpp1
      Start  9: test.debug_prefix_map
      Start 10: test.depend
      Start 11: test.direct
      Start 12: test.direct_gcc
      Start 13: test.fileclone
      Start 14: test.hardlink
      Start 15: test.inode_cache
      Start 16: test.input_charset
      Start 17: test.ivfsoverlay
      Start 18: test.masquerading
      Start 19: test.modules
      Start 20: test.multi_arch
      Start 21: test.namespace
      Start 22: test.no_compression
      Start 23: test.nocpp2
      Start 24: test.nvcc
 1/44 Test  #1: unittest .........................   Passed    0.07 sec
      Start 25: test.nvcc_direct
 2/44 Test #20: test.multi_arch ..................***Skipped   0.18 sec
      Start 26: test.nvcc_ldir
 3/44 Test  #6: test.color_diagnostics ...........***Skipped   0.19 sec
      Start 27: test.nvcc_nocpp2
 4/44 Test #13: test.fileclone ...................***Skipped   0.19 sec
      Start 28: test.pch
 5/44 Test #24: test.nvcc ........................***Skipped   0.18 sec
      Start 29: test.profiling
 6/44 Test #17: test.ivfsoverlay .................***Skipped   0.19 sec
      Start 30: test.profiling_clang
 7/44 Test  #7: test.config ......................   Passed    0.20 sec
      Start 31: test.profiling_gcc
 8/44 Test #19: test.modules .....................***Skipped   0.19 sec
      Start 32: test.profiling_hip_clang
 9/44 Test #25: test.nvcc_direct .................***Skipped   0.19 sec
      Start 33: test.readonly
10/44 Test  #9: test.debug_prefix_map ............***Failed    0.32 sec
Compiler:         gcc (/nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/gcc)
Compiler version: gcc (GCC) 10.3.0
CUDA compiler:    not available

Running test suite debug_prefix_map./build/source/test/run: line 156: /nix/store/g876mcfvkd6h80s28z5p1czvdcsxr3q3-i686-unknown-linux-gnu-binutils-2.35.2/bin/objdump: No such file or directory
/build/source/test/run: line 156: /nix/store/g876mcfvkd6h80s28z5p1czvdcsxr3q3-i686-unknown-linux-gnu-binutils-2.35.2/bin/objdump: No such file or directory

FAILED

Test suite:     debug_prefix_map (line 34)
Test case:      Mapping of debug info CWD
Failure reason: File test.o does not contain "some_name_not_likely_to_exist_in_path"

Actual statistics counters
==========================
autoconf_test                    0
bad_compiler_arguments           0
bad_output_file                  0
cache_miss                       1
cache_size_kibibyte              0
called_for_link                  0
called_for_preprocessing         0
cleanups_performed               0
compile_failed                   0
compiler_check_failed            0
compiler_produced_empty_output   0
compiler_produced_no_output      0
compiler_produced_stdout         0
could_not_find_compiler          0
could_not_use_modules            0
could_not_use_precompiled_header 0
direct_cache_hit                 0
direct_cache_miss                1
error_hashing_extra_file         0
files_in_cache                   2
internal_error                   0
missing_cache_file               0
multiple_source_files            0
no_input_file                    0
output_to_stdout                 0
preprocessed_cache_hit           0
preprocessed_cache_miss          1
preprocessor_error               0
primary_storage_hit              0
primary_storage_miss             2
recache                          0
secondary_storage_error          0
secondary_storage_hit            0
secondary_storage_miss           0
secondary_storage_timeout        0
unsupported_code_directive       0
unsupported_compiler_option      0
unsupported_source_language      0

Test data and log file have been left in testdir/3325 / testdir/failed

      Start 34: test.readonly_direct
11/44 Test #26: test.nvcc_ldir ...................***Skipped   0.18 sec
      Start 35: test.sanitize_blacklist
12/44 Test #27: test.nvcc_nocpp2 .................***Skipped   0.19 sec
      Start 36: test.secondary_file
13/44 Test  #8: test.cpp1 ........................   Passed    0.39 sec
      Start 37: test.secondary_http
14/44 Test #32: test.profiling_hip_clang .........***Skipped   0.19 sec
      Start 38: test.secondary_redis
15/44 Test #30: test.profiling_clang .............***Skipped   0.20 sec
      Start 39: test.secondary_url
16/44 Test #16: test.input_charset ...............   Passed    0.42 sec
      Start 40: test.serialize_diagnostics
17/44 Test #18: test.masquerading ................   Passed    0.49 sec
      Start 41: test.source_date_epoch
18/44 Test #28: test.pch .........................***Skipped   0.34 sec
      Start 42: test.split_dwarf
19/44 Test #35: test.sanitize_blacklist ..........***Skipped   0.22 sec
      Start 43: test.stats_log
20/44 Test #38: test.secondary_redis .............***Skipped   0.20 sec
      Start 44: test.upgrade
21/44 Test #37: test.secondary_http ..............***Skipped   0.21 sec
22/44 Test #40: test.serialize_diagnostics .......***Skipped   0.21 sec
23/44 Test #21: test.namespace ...................   Passed    0.64 sec
24/44 Test #39: test.secondary_url ...............   Passed    0.28 sec
25/44 Test #22: test.no_compression ..............   Passed    0.68 sec
26/44 Test #34: test.readonly_direct .............   Passed    0.44 sec
27/44 Test #33: test.readonly ....................   Passed    0.52 sec
28/44 Test #15: test.inode_cache .................   Passed    0.81 sec
29/44 Test #14: test.hardlink ....................   Passed    0.82 sec
30/44 Test #43: test.stats_log ...................   Passed    0.25 sec
31/44 Test #44: test.upgrade .....................   Passed    0.25 sec
32/44 Test  #4: test.cache_levels ................   Passed    0.87 sec
33/44 Test #41: test.source_date_epoch ...........   Passed    0.61 sec
34/44 Test #12: test.direct_gcc ..................   Passed    1.19 sec
35/44 Test #31: test.profiling_gcc ...............   Passed    1.02 sec
36/44 Test #42: test.split_dwarf .................   Passed    0.89 sec
37/44 Test #36: test.secondary_file ..............   Passed    1.12 sec
38/44 Test #10: test.depend ......................   Passed    1.56 sec
39/44 Test #29: test.profiling ...................   Passed    1.59 sec
40/44 Test  #5: test.cleanup .....................   Passed    2.02 sec
41/44 Test  #3: test.basedir .....................   Passed    2.41 sec
42/44 Test  #2: test.base ........................   Passed    6.97 sec
43/44 Test #23: test.nocpp2 ......................   Passed    6.97 sec
44/44 Test #11: test.direct ......................   Passed    7.01 sec

98% tests passed, 1 tests failed out of 44

Total Test time (real) =   7.02 sec

The following tests did not run:
	  6 - test.color_diagnostics (Skipped)
	 13 - test.fileclone (Skipped)
	 17 - test.ivfsoverlay (Skipped)
	 19 - test.modules (Skipped)
	 20 - test.multi_arch (Skipped)
	 24 - test.nvcc (Skipped)
	 25 - test.nvcc_direct (Skipped)
	 26 - test.nvcc_ldir (Skipped)
	 27 - test.nvcc_nocpp2 (Skipped)
	 28 - test.pch (Skipped)
	 30 - test.profiling_clang (Skipped)
	 32 - test.profiling_hip_clang (Skipped)
	 35 - test.sanitize_blacklist (Skipped)
	 37 - test.secondary_http (Skipped)
	 38 - test.secondary_redis (Skipped)
	 40 - test.serialize_diagnostics (Skipped)

The following tests FAILED:
	  9 - test.debug_prefix_map (Failed)
Errors while running CTest
```

### Notify maintainers

@kira-bruneau  @r-burns 

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.81, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5pre20211126_55275fc`
 - channels(root): `"nixos-21.05.2549.a1007637cea, home-manager-21.05, nixos-unstable-21.11pre304920.dd98b100651"`
 - channels(giulio): `""`
 - nixpkgs: `/nix/store/wnmzij43q35h8vil2apilxzkfnjcdj4p-source`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by kira-bruneau** [comment]
I don't have any problems building `nix-build -A pkgsCross.gnu32.ccache` since the tests are disabled in a cross compilation build. Are you doing anything extra to enable them?

It seems like the failure can be fixed by replacing `binutils.bintools` with `buildPackages.binutils.bintools`, but I'm not sure how you've enabled the tests, so I can't verify if it works.

**Comment by r-burns** [comment]
Yes, I'm surprised the tests are running at all in a cross-compilation scenario. Maybe you are already on a 32-bit build machine, but forcing the cross-compilation pathway with pkgsCross?

Also, unrelated to this specific issue, if you are trying to build 32-bit binaries on an x86 host you may have better luck with `pkgsi686Linux`, which tends to be more reliable as it builds natively.

**Comment by peperunas** [comment]
Well, I did a quick test using `ccache` in this configuration. I am not using it anymore and, as you mentioned, it probably was some kind of misconfiguration on my side. I will re-open the issue in case this is actually an issue.

As a side note, @r-burns that's what I was doing but I had to cross-compile in the end. Thanks for the suggestion!

Sorry for bothering you and thank you for your prompt response!
